src/Services/QueuedJobHandler.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Symbiote\QueuedJobs\Services;
+
+use Monolog\Handler\AbstractProcessingHandler;
+use SilverStripe\Core\Injector\Injectable;
+use Symbiote\QueuedJobs\DataObjects\QueuedJobDescriptor;
+
+/**
+ * Writes log output to a job descriptor
+ */
+class QueuedJobHandler extends AbstractProcessingHandler
+{
+    use Injectable;
+
+    /** @var QueuedJob */
+    protected $job;
+
+    /** @var QueuedJobDescriptor */
+    protected $jobDescriptor;
+
+    public function __construct(QueuedJob $job, QueuedJobDescriptor $jobDescriptor)
+    {
+        parent::__construct($job, $jobDescriptor);
+
+        $this->job = $job;
+        $this->jobDescriptor = $jobDescriptor;
+    }
+
+    /**
+     * @return QueuedJob
+     */
+    public function getJob()
+    {
+        return $this->job;
+    }
+
+    /**
+     * @return QueuedJobDescriptor
+     */
+    public function getJobDescriptor()
+    {
+        return $this->jobDescriptor;
+    }
+
+    /**
+     * Writes the record down to the log of the implementing handler
+     *
+     * @param  array $record
+     * @return void
+     * @throws \SilverStripe\ORM\ValidationException
+     */
+    protected function write(array $record)
+    {
+        $this->job->addMessage($record['message']);
+        $this->jobDescriptor->SavedJobMessages = serialize($this->job->getJobData()->messages);
+        $this->jobDescriptor->write();
```

Comment:
@bergice That's one database write per log entry. Which can be just as expensive as the task it's actually running. In the case which triggered this, that would be moving potentially tens of thousands of files. I think we just made this task sustantially slower by introducing logging. Your local database won't be a good test case for this, remove databases have at least 1ms roundtrip. Have you looked at Monolog-level batching?

Reply:
Good point, I've implemented this using `BufferHandler`.